examples/demo-simple-stf/src/lib.rs

```diff
@@ -26,7 +33,7 @@ pub enum ApplySlotResult {
     Success,
 }
 
-impl<Vm: Zkvm, Cond: ValidityCondition, B: BlobReaderTrait> StateTransitionFunction<Vm, B>
+impl<Vm: ZkVerifier, Cond: ValidityCondition, Da: DaSpec> StateTransitionFunction<Vm, Da>
     for CheckHashPreimageStf<Cond>
```

Comment:
```suggestion
    for CheckHashPreimageStf<DA::ValidityCondition>
```

Reply:
This would also ensure that used validity condition are the same type as in DaSpec